drone/src/drone/agent/mod.rs

```diff
@@ -132,24 +154,19 @@ pub async fn run_agent(agent_opts: AgentOptions) -> Result<()> {
             cluster: cluster.clone(),
             ip,
         };
-        do_with_retry(
-            || nats.request(&request),
-            30,
-            Duration::from_secs(10),
-        )
-        .await?
+        do_with_retry(|| nats.request(&request), 30, Duration::from_secs(10)).await?
     };
 
     match result {
         DroneConnectResponse::Success { drone_id } => {
-            {
-                let nats = nats.clone();
-                let cluster = cluster.clone();
-                tokio::spawn(ready_loop(nats, drone_id, cluster));
-            }
-
+            let executor = Arc::new(Executor::new(docker, db, nats.clone()));
```

Comment:
Executor is designed to be cloned (the underlying types use `Arc`), so wrapping it in an `Arc` creates another layer of indirection. Better to just create an executor and clone it.

Reply:
needed to do a small refactor here and there to get his to work, but done!